### DISCUSSION TITLE
adding interactive marble diagrams

### DISCUSSION THREAD

**Discussion by Ayudh** [ReactiveX/rxjs#6258] [Ideas / Feature request]
In current docs, we have a PNG marble diagram to help the developer to understand about an operator in a glance.
I am thinking it will be more helpful to add interactive marble diagrams (similar to [RxMarbles](https://rxmarbles.com/#switchMap)).
With this, dev can easily understand the edge cases and compare operators ([switchMap](https://rxmarbles.com/#switchMap) vs [mergeMap](https://rxmarbles.com/#mergeMap)).

Please suggest if this can be considered. I am willing to work on this feature.

**Comment by Ayudh** [comment]
@cartant @benlesh any thoughts on this?

**Reply by cartant** [reply]
Getting the documentation for version 7 ready for a release is the priority, right now. This could be given some consideration after version 7 is released, but, ATM, I don't think anyone has the bandwidth to discuss this.

**Reply by Ayudh** [reply]
Thanks for the response. we can discuss this post v7 release.

**Comment by Ayudh** [comment]
@cartant @benlesh can we consider this? Please share your thoughts

If yes, I will try to implement a POC.

**Reply by benlesh** [reply]
The real problem with anything too advanced in the documentation is that we're then on the hook for maintaining it. I'm hesitant to add any more complexity than what's already there.

**Reply by Ayudh** [reply]
Understood. On the other side, this helps developers to understand/compare an operator with edge cases.
I think this is similar to the marble diagram PNG we have currently today.

Please share your thoughts so we can conclude this
